src/kani-driver/src/call_cbmc.rs

```diff
@@ -127,3 +133,192 @@ impl KaniSession {
         args
     }
 }
+
+/// Solve Unwind Value from conflicting inputs of unwind values. (--unwind, annotation-unwind, --harness-unwind)
+pub fn resolve_unwind_value(args: &KaniArgs, harness_metadata: &HarnessMetadata) -> Option<u32> {
+    // Check for which flag is being passed and prioritize extracting unwind from the
+    // respective flag/annotation.
+    if let Some(harness_unwind) = args.harness_unwind {
+        Some(harness_unwind)
+    } else if let Some(annotation_unwind) = harness_metadata.unwind_value {
+        Some(annotation_unwind)
+    } else if let Some(default_unwind) = args.unwind {
+        Some(default_unwind)
+    } else {
+        None
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::args;
+    use crate::metadata::{find_proof_harness, mock_proof_harness};
+    use structopt::StructOpt;
+
+    use super::*;
+
+    #[test]
+    fn check_default_annotation_unwind_resolve() {
+        // Annotation value for unwind takes precedence over unwind value
+        // except when there is no #[kani::unwind()] provided.
+        let args: Vec<OsString> =
+            ["kani", "--no-default-checks", "--unwind", "2"].iter().map(|&s| s.into()).collect();
+        let unwind_args = args::KaniArgs::from_iter(args);
+
+        let harnesses = vec![
+            mock_proof_harness("check_one", Some(1)),
+            mock_proof_harness("module::check_two", Some(3)),
+            mock_proof_harness("module::not_check_three", None),
+        ];
+
+        let merged_1 = resolve_unwind_value(
+            &unwind_args,
+            find_proof_harness("check_one", &harnesses).unwrap(),
+        );
+        let merged_2 = resolve_unwind_value(
+            &unwind_args,
+            find_proof_harness("check_two", &harnesses).unwrap(),
+        );
+        let merged_3 = resolve_unwind_value(
+            &unwind_args,
+            find_proof_harness("not_check_three", &harnesses).unwrap(),
+        );
+
+        assert!(merged_1 == Some(1));
+        assert!(merged_2 == Some(3));
+        assert!(merged_3 == Some(2));
+    }
+
+    #[test]
+    fn check_harness_annotation_unwind_resolve() {
+        // Command line unwind value for specific harnesses take precedence over unwind annotation value
+        let args_1: Vec<OsString> =
```

Comment:
These tests are over-complex. The behavior of `resolve_unwind_value` does not change with different `--harness` values, so checking multiple doesn't test anything here